src/applications/claims-status/components/claim-status-tab/RecentActivity.jsx

```diff
@@ -3,76 +3,163 @@ import PropTypes from 'prop-types';
+        return 'There was an update to this item';
+    }
+  };
 
-const getSortedItems = claim => {
-  // Get items from trackedItems and claimPhaseDates
-  const trackedItems = generateTrackedItems(claim);
-  const phaseItems = [];
-  const items = [...trackedItems, ...phaseItems];
+  const getPhaseItemDescription = (currentPhaseBack, phase) => {
+    const phaseItemText = getPhaseItemText(phase, cstClaimPhasesEnabled);
+    switch (phase) {
+      case 1:
+      case 8:
+        return phaseItemText;
+      case 2:
+      case 7:
+        return `Your claim moved into ${phaseItemText}`;
+      case 3:
+      case 4:
+      case 5:
+      case 6:
+        if (currentPhaseBack) {
+          return `Your claim moved back to ${phaseItemText}`;
+        }
+        return `Your claim moved into ${phaseItemText}`;
+      default:
+        return '';
+    }
+  };
 
-  return items.sort((item1, item2) => {
-    return moment(item2.date) - moment(item1.date);
-  });
-};
+  const generateTrackedItems = () => {
+    const { trackedItems } = claim.attributes;
+
+    return trackedItems.map(item => ({
+      id: item.id,
+      date: getTrackedItemDateFromStatus(item),
+      description: getTrackedItemDescription(item),
+      displayName: item.displayName,
+      status: item.status,
+      type: 'tracked_item',
+    }));
+  };
+
+  const isEventOrPrimaryPhase = event => {
+    if (event.type === 'phase_entered') {
+      return event.phase <= 3 || event.phase >= 7;
+    }
+
+    return !!getItemDate(event);
+  };
+
+  const generatePhaseItems = () => {
+    const {
+      currentPhaseBack,
+      previousPhases,
+    } = claim.attributes.claimPhaseDates;
+    const phases = [];
+
+    const regex = /\d+/;
+
+    // Add phase dates using the complete date of other phases
+    const phaseKeys = Object.keys(previousPhases);
+    phaseKeys.forEach(phaseKey => {
+      const phase = Number(phaseKey.match(regex)[0]) + 1;
+      phases.push({
+        id: phase,
+        type: 'phase_entered',
+        // We are assuming here that each phaseKey is of the format:
+        // phaseXCompleteDate, where X is some integer between 1 and 7
+        // NOTE: Adding 1 because the a phases completion date is
+        // analagous to the phase after it's start date
+        // eg. phase1CompleteDate = start date for phase 2
+        phase,
+        description: getPhaseItemDescription(currentPhaseBack, phase),
+        date: previousPhases[phaseKey],
+      });
+    });
+
+    // Add initial phase date since its not inculded in previousPhases
+    phases.push({
+      id: 1,
+      type: 'filed',
+      phase: 1,
+      description: getPhaseItemDescription(currentPhaseBack, 1),
+      date: claim.attributes.claimDate,
+    });
+
+    // When cst_claim_phases is enabled then we remove steps 4-6 and only keep step 3
+    return cstClaimPhasesEnabled
+      ? phases
+      : phases.filter(isEventOrPrimaryPhase);
+
+    // const currentPhase = getPhaseFromStatus(
```

Comment:
@jerekshoe this logic was in the timeline...I was confused on if it was necessary...from what I can see I dont think I need it but wanted to have a convo with you before I fully removed it!